src/instructlab/model/evaluate.py

```diff
@@ -1,52 +1,145 @@
 )
 import click
 
-benchmark_names_to_classes = frozenset({
-    "mmlu": MMLU_Evaluator,
-    "mt_bench": MT_Bench_Evaluator,
-    "pr_bench": PR_Bench_Evaluator,
-    "pr_mmlu": PR_MMLU_Evaluator,
-})
+BENCHMARK_TO_CLASS_MAP = frozenset(
+    {
+        "mmlu": MMLUEvaluator,
+        "mmlu_branch": MMLUBranchEvaluator,
+        "mt_bench": MTBenchEvaluator,
+        "mt_bench_branch": MTBenchBranchEvaluator,
+    }
+)
+
+
+def get_evaluator(
+    model_name,
+    benchmark,
+    judge_model_name,
+    output_dir,
+    max_workers,
+    taxonomy_path,
+    branch,
+    tasks,
+    few_shots,
+    batch_size,
+    task,
+    sdg_path,
+) -> Evaluator:
+    """takes in arguments from the CLI and uses 'benchmark' to validate other arguments
+    if all needed configuration is present, returns the appropriate Evaluator class for the benchmark
+    otherwise raises an exception for the missing values
+    """
+
+    # ensure skills benchmarks have proper arguments if selected
+    if benchmark in ["mt_bench", "mt_bench_branch"]:
+        required_args = [
+            model_name,
+            judge_model_name,
+            output_dir,
+            max_workers,
+        ]
+        if benchmark == "mt_bench_branch":
+            required_args.append(taxonomy_path, branch)
+        if any(required_args) is None:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_args}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        else:
+            evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+            if benchmark == "mt_bench":
+                return evaluator_class(
+                    model_name, judge_model_name, output_dir, max_workers
+                )
+            else:
+                return evaluator_class(
+                    model_name,
+                    judge_model_name,
+                    taxonomy_path,
+                )
+
+    # ensure knowledge benchmarks have proper arguments if selected
+    if benchmark in ["mmlu", "mmlu_branch"]:
+        required_args = [few_shots, batch_size]
+        if benchmark == "mmlu":
+            required_args.append(tasks)
+        if benchmark == "mmlu_branch":
+            required_args.extend([task, sdg_path])
+        if any(required_args) is None:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_args}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        else:
+            evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+            if benchmark == "mmlu":
+                return evaluator_class(tasks, few_shots, batch_size)
+            else:
+                return evaluator_class(sdg_path, task, few_shots, batch_size)
 
 
 @click.command(cls=DYMGroup)
+@click.option(
+    "--model-name",
+    type=click.Path(),
+    help="Path to model to be evaluated",
```

Comment:
Name of model to be evaluated?

Reply:
Where will it look for the model once it has the name?

Reply:
For mtbench, it looks for the model being served on server_url.  The mmlu work hasn't switched to looking up the hosted model yet.  My comment was just meant to point out the description didn't match the option name.

Reply:
@alinaryan ^^ FYI